bin/cron/push_latest_rds_backup_to_secondary_account

```diff
@@ -0,0 +1,127 @@
+  shared_snapshots = []
+  attempts = 0
+  max_attempts = 60
+  until shared_snapshots.count > 0
+    sleep(1)
+    attempts += 1
+    if attempts > max_attempts
+      raise SnapshotNotSharedError, 'Could not find shared snapshot after 60 seconds'
+    end
+    shared_snapshots = rds_backup.db_snapshots({
+      max_records: 100,
+      include_shared: true,
+    }).select { |snap|
+      snap.snapshot_id.start_with?('arn:aws:rds:') && snap.snapshot_id.end_with?(temp_snapshot_name)
+    }
+  end
+  shared_snapshots.first
+end
+
+def share_snapshot_with_account(credentials_backup, latest_snapshot)
+  # Copy the automated backup into a shareable manual one
+  temp_snapshot_name = 'temp-snapshot-' + Time.now.to_i.to_s
+  copied_snapshot = latest_snapshot.copy({ target_db_snapshot_identifier: temp_snapshot_name })
+
+  wait_for_snapshot_copy(copied_snapshot)
+  puts "Temporary snapshot #{copied_snapshot.snapshot_id} is ready - #{Time.now}"
+
+  # Share the new snapshot with the backup account
+  backup_account_id = get_aws_account_id(credentials_backup).first
+  restore_attribute = copied_snapshot.attributes.find { |attr| attr.name == 'restore' }
+  restore_attribute.modify({
+      values_to_add: [backup_account_id],
+      values_to_remove: []
+  })
+  return copied_snapshot, temp_snapshot_name
+end
+
+def main
+  HipChat.developers 'Beginning cross-account RDS backup'
+  credentials_primary = Aws::SharedCredentials.new(profile_name: 'default')
+  rds_primary = Aws::RDS::Resource.new(credentials: credentials_primary)
+  credentials_backup = Aws::SharedCredentials.new(profile_name: 'backup')
+  rds_backup = Aws::RDS::Resource.new(credentials: credentials_backup)
+
+  production_instance = rds_primary.db_instances.find{ |i| i.id == 'production' }
+
+  # Find the latest automated backup
+  sorted_snapshots = production_instance.snapshots.
+      select{ |snap| snap.status == 'available' && !snap.snapshot_id.start_with?('temp-snapshot-') }.
+      sort{ |a, b| a.snapshot_create_time <=> b.snapshot_create_time }
+  latest_snapshot = sorted_snapshots.last
+
+  begin
+    copied_snapshot, temp_snapshot_name = share_snapshot_with_account(credentials_backup, latest_snapshot)
+    shared_snapshot = find_shared_snapshot_on_backup(rds_backup, temp_snapshot_name)
+
+    # On the backup account, copy the shared snapshot
+    begin
+      backed_up_snapshot = shared_snapshot.copy({
+        target_db_snapshot_identifier: latest_snapshot.snapshot_id.sub('rds:', '')
+      })
+      wait_for_snapshot_copy(backed_up_snapshot)
+      HipChat.developers "Completed cross account RDS backup: #{latest_snapshot.snapshot_id}"
+    rescue Aws::RDS::Errors::DBSnapshotAlreadyExists
+      HipChat.developers "RDS snapshot backup error: #{latest_snapshot.snapshot_id} already backed up"
+    end
+  ensure
+    # Clean up the temporary snapshot
+    puts "Deleting temporary backup snapshot #{copied_snapshot.snapshot_id}"
+    copied_snapshot.delete
+  end
+end
+
```

Comment:
Though probably not necessary here, I'll note the `only_one_running` method (https://github.com/code-dot-org/code-dot-org/blob/staging/lib/cdo/only_one.rb) that is often used in our cron jobs.